### DISCUSSION TITLE
matchPackageNames -- what does it match? packageName? depName?

### DISCUSSION THREAD

**Discussion by chrono** [renovatebot/renovate#20213] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.108.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate end users (me and my colleagues) having a clear understanding of what a depName, a packageName, and their relationship to each other and their matchers is.

### What you tried so far.

Reading documentation, examples, and using renovate.

Consider this (working) config file:

```
module.exports = {
  "enabledManagers": ["regex", "asdf"],
  "regexManagers": [
    {
      "description": "Update a docker image version referencing terraform in a yaml file",
      "fileMatch": ["pipeline\\.yml$"],
      "matchStrings": [
        "infra-terraform-test:(?<currentValue>.+?)\"\\s"
      ],
      "versioningTemplate": "semver",
      "datasourceTemplate": "github-releases",
      "depNameTemplate": "terraform",
      "packageNameTemplate": "hashicorp/terraform"
    }
  ],
  "packageRules": [
      {
          "matchManagers": ["regex"],
          "matchPackageNames": ["terraform"],
          "extractVersion": "^v(?<version>.*)$"
      }
  ]
}
```

and this `.tool-versions` file for the `asdf` manager:
```
terraform 0.14.7
```

I looked at the debug output from the asdf manager to discover the depName, datasource and packageName used for terraform and based my regex manager on that (I want both changes to happen in the same PR). Now I want to apply `extractVersion` to this. But I don't want to place it into the regex manager [for the purpose of the example]. Let's say I want to use a package rule to modify this.

In order to make it work, I had to use the string `terraform` (not `hashicorp/terraform`) in the `matchPackageNames` list. I'm confused, because in the regex manager, I clearly declare `depName=terraform` and `packageName=hashicorp/terraform`.

There are other [Q&A discussion questions asking about this](https://github.com/renovatebot/renovate/discussions/14885) and the documentation simply says [matchPackageNames: Package names to match. Valid only within a packageRules object.](https://docs.renovatebot.com/configuration-options/#matchpackagenames).

Is the documentation incomplete? Should I perhaps file a documentation bug?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
You have stumbled upon a little technical debt hole we accidentally for ourselves.

In short, we started using the name `matchPackageNames` but all along it's been matching against `depName`. Now it's a little hard for us to switch without risking breaking some people.

We've now added `matchDepNames` which ONLY matches depName.

By default, `matchPackageNames` does the same. But for self-hosted (which you've said you're not) we have this experimental option to ALSO match packageName: https://github.com/renovatebot/renovate/blob/f754f8a010abaa22e52b0e7642b01b76b5f9a0be/lib/util/package-rules/package-names.ts#L21-L26

I think we might want to take the hard step of modifying behavior in the next major release. Any PRs you would like to submit against documentation would be welcome (much better and probably faster even than a bug report). Docs are all in this repo
